src/config.ts

```diff
@@ -43,6 +44,11 @@ export interface Script {
   type: string | null;
 }
 
+/**
+ * A function that generates a page fragment such as head, header or footer.
+ */
+type PageFragmentFunction = (title: string | null, data: FrontMatter, path: string) => string;
```

Comment:
Would rather use named arguments here (`({title, data, path})`), not positional (`(title, data, path)`), so that it’s easier for us to add more arguments in the future without it becoming unwieldy.

Also this type is referenced by `Config` and hence should be exported.